### PR TITLE
Maintain Pathways filters state in the path query params

### LINKED ISSUE #1310: Ensure URL path includes state, Pathways/Practices, page, chart, filters selected
**Is your feature request related to a problem? Please describe.**
A clear and concise description of what the problem is, e.g. we don't currently convey...

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**
Add any other context about the feature request here.

### PR DESCRIPTION
**Pull request by jovergaag**
## Description of the change

Updates the path query params when the Pathways filters change. Currently, the filters persist across page navigations, only the enabled Filters are included in the query params. 

Also updates the path query params to include the `currentMetricId` on the Practices page.

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #1310

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally
- [ ] E2E have been run for Lantern changes ([Steps in the Readme](https://github.com/Recidiviz/pulse-dashboard#running-e2e-tests))

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

File: src/core/CoreStore/PagePracticesStore.ts [hunk 1]
```diff
@@ -79,7 +83,7 @@ export default class PagePracticesStore {
   }
 
   setSelectedMetricId(metricId: MetricType): void {
-    this.selectedMetricId = metricId;
+    this.selectedMetricId = metricId || METRIC_TYPES.OVERALL;
```
Comment (Catacola) [inline src/core/CoreStore/PagePracticesStore.ts:86]: Can `metricId` be falsy?

File: src/core/PathwaysFilterBar/PathwaysFilterBar.tsx [hunk 2]
```diff
@@ -14,29 +14,50 @@
...
+const PathwaysFilterBar: React.FC<{
   filterOptions: PopulationFilters;
   handleDownload: () => Promise<void>;
   enabledFilters?: EnabledFilters;
 }> = ({ filterOptions, handleDownload, enabledFilters = [] }) => {
   const { filtersStore } = useCoreStore();
   const { filters } = filtersStore;
 
+  // if current query params do not match enabled filters, update query
+  const [query, setQuery] = useQueryParams(filterQueryParams);
```
Comment (Catacola) [inline src/core/PathwaysFilterBar/PathwaysFilterBar.tsx:51]: Whoever wrote this package deserves a cookie

**Review by Catacola — APPROVED** [review]
This is fantastic and works so well!

I just want to flag that since we use different labels for the filters than in the data, the option in the URL doesn't always match the dropdown. For example, on the prison, the legal status dropdown will say Rider whereas the URL will say `TREATMENT_IN_PRISON`. This isn't a blocker, but we may want to align the data coming from the backend with the copy in the filter bar.

File: src/core/PathwaysFilterBar/PathwaysFilterBar.tsx (hunk 2 above)
Comment (jovergaag) [reply src/core/PathwaysFilterBar/PathwaysFilterBar.tsx:51]: oh snap gingersnap

File: src/core/CoreStore/PagePracticesStore.ts (hunk 1 above)
Comment (jovergaag) [reply src/core/CoreStore/PagePracticesStore.ts:86]: Yes, when we are setting the metric is in `withRouteSync` when first landing on the Practices page the params are empty so we have an undefined metricId. In this case we want to revert to default.

**Comment by sychang** [comment]
Thanks for tagging me in! We will want to align the BE and FE strings, especially since the URL param is so visible to the user. The phrases can be pretty jarring for users if the terminology we're showing misaligns with what they use.

**Comment by jovergaag** [comment]
> Thanks for tagging me in! We will want to align the BE and FE strings, especially since the URL param is so visible to the user. The phrases can be pretty jarring for users if the terminology we're showing misaligns with what they use.

@sychang Is this launch blocking?

**Comment by sychang** [comment]
Not launch blocking for ID, will be for future states

**Pull request merged**